cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonLanguageFrontend.kt

```diff
@@ -326,6 +324,112 @@ class PythonLanguageFrontend(language: Language<PythonLanguageFrontend>, ctx: Tr
         return tud
     }
 
+    private fun getRootPath(file: File, rootPaths: List<Path>): Path {
+        rootPaths.forEach {
+            if (file.path.removePrefix(it.toString()) != file.path.toString()) return it
+        }
+        return file.toPath().parent
+    }
+
+    override fun gatherExternalSources(
+        rootPaths: List<Path>,
+        source: File,
+        externalSources: MutableList<File>,
+        processedImports: MutableList<String>,
+    ): List<File> {
+
+        val pythonExternalSources =
+            externalSources
+                .filter { externalFile ->
+                    language.fileExtensions.any { externalFile.path.endsWith(it) }
+                }
+                .toMutableList()
+
+        val importRe = "(?m)^(?:from[ ]+(\\S+)[ ]+)?import[ ]+(\\S+)[ ]*\$".toRegex()
```

Comment:
I object having this regex here.
- It's not documented.
- I don't like parsing the sources manually with a regex -> can't we use the `ast.parse` result for this?
- Can we please name groups, if we have to use a regex? This would make it much easier to understand what's going on.
- What about `import foo` without the `from` part?
- What about `as` in the import?

Reply:
I removed the regex approach.